FaunaDB.Client/Types/Attributes.cs

```diff
@@ -0,0 +1,122 @@
+    ///
+    /// class Product
+    /// {
+    ///     private string description;
+    ///     private double price;
+    ///
+    ///     [Creator]
+    ///     public Product([Field("Description")] string description, [Field("Price")] double price)
+    ///     {
+    ///         this.description = description;
+    ///         this.price = price;
+    ///     }
+    /// }
+    /// </example>
+    [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property | AttributeTargets.Parameter)]
+    public class FieldAttribute : Attribute
+    {
+        /// <summary>
+        /// The name of the property when encoding/decoding objects.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        /// The default value used for a missing property when decoding an object.
+        /// </summary>
+        public object DefaultValue { get; set; }
+
+        public FieldAttribute(string name)
+        {
+            Name = name;
+        }
+    }
+
+    /// <summary>
+    /// Instruct the encoder to not encode the specified member.
+    /// </summary>
+    /// <example>
+    /// class User
+    /// {
+    ///     [Field("user_name")]
+    ///     public string UserName { get; set; }
+    ///
+    ///     [Ignore]
+    ///     public string Password { get; set; }
+    /// }
+    ///
+    /// var user = new User { UserName = "john", Password = "s3cr3t" };
+    ///
+    /// var encoded = Encoder.Encode(user);
+    ///
+    /// //encoded will be equivalent to:
+    ///
+    /// ObjectV.With("user_name", "john")
+    /// </example>
+    [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property)]
+    public class IgnoreAttribute : Attribute
+    { }
+
+    /// <summary>
+    /// Instruct the decoder wich constructor to use when decoding a object.
+    /// It can also be used in a public static method instead in a constructor.
+    /// That attribute can only be used once per class.
+    /// </summary>
+    /// <example>
+    /// class Product
+    /// {
+    ///     public string Description { get; set; }
+    ///     public double Price { get; set; }
+    ///
+    ///     [Creator]
+    ///     public Product(string description, double price)
+    ///     {
+    ///         Description = description;
+    ///         Price = price;
+    ///     }
+    /// }
+    ///
+    /// class Order
+    /// {
+    ///     public string Number { get; set; }
+    ///     public List&lt;Product&gt; Products { get; set; }
+    ///
+    ///     [Creator]
+    ///     public static Order CreateOrder(string number, List&lt;Product&gt; products)
+    ///     {
+    ///         return new Order
+    ///         {
+    ///             Number = number,
+    ///             Products = products
+    ///         }
+    ///     }
+    /// }
+    /// </example>
+    [AttributeUsage(AttributeTargets.Constructor | AttributeTargets.Method)]
+    public class CreatorAttribute : Attribute
```

Comment:
IDK, I was taking a look on [Json.NET attributes ](http://www.newtonsoft.com/json/help/html/serializationattributes.htm) and I started to wonder if we should take step back and prefix our attributes with something like `Fauna`, ex: `FaunaField`, `FaunaIgnore`, `FaunaConstructor`? I'm saying that because just `Field` or `Ignore` already caused some names collision, specially `Field` because we already have a class with that name. Maybe @eaceaser can help us on this